PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py

```diff
@@ -239,7 +239,16 @@ def miniAOD_customizeCommon(process):
     process.load('RecoJets.JetProducers.QGTagger_cfi')
     task.add(process.QGTaggerTask)
 
-    process.patJets.userData.userFloats.src += [ cms.InputTag('QGTagger:qgLikelihood'), ]
+    process.patJets.userData.userFloats.src += [ 'QGTagger:qgLikelihood', ]
+
+    #HF jet shower shape
+    process.load('RecoJets.JetProducers.hfJetShowerShape_cfi')
+    task.add(process.hfJetShowerShape)
+    process.hfJetShowerShape.jets = cms.InputTag("ak4PFJetsCHS")
+    process.hfJetShowerShape.vertices = cms.InputTag("offlinePrimaryVertices")
```

Comment:
- try to not modify values from loaded objects where possible
- related to https://github.com/cms-sw/cmssw/pull/31271#discussion_r482564203 : would it make more sense to use `offlineSlimmedPrimaryVertices` to improve reproducibility?

Reply:
To make sure I get that right: you suggest to set default values for jets and vertices in the .cc class right? 
Also, offlineSlimmedPrimaryVertices is not an AOD collection so that means one would rely on the fact that offlineSlimmedPrimaryVertices is produced before in the MINIAOD step. Is that ok to do? It naively feels a bit dangerous. Thanks !

Reply:
@slava77

Reply:
> To make sure I get that right: you suggest to set default values for jets and vertices in the .cc class right?

yes; 
https://github.com/cms-sw/cmssw/pull/31271#pullrequestreview-483515551:
As it stands then, hfJetShowerShape is the only running instance of the module out of all 3 options, which means that the definitions of hfJetShowerShapeDefault is not necessary and can be replaced by the earlier available hfJetShowerShape.

> Also, offlineSlimmedPrimaryVertices is not an AOD collection so that means one would rely on the fact that offlineSlimmedPrimaryVertices is produced before in the MINIAOD step. Is that ok to do? It naively feels a bit dangerous. Thanks !

If there is not dependency of slimmed PVs on patJets (where `hfJetShowerShape` is consumed), I think that it should be fine to use them